### ISSUE TITLE
508-defect-2 [SCREENREADER] VA needs to have aria-label to properly pronounce it

### ISSUE THREAD

**Issue by noahgelman** [department-of-veterans-affairs/va.gov-team#20238]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Noah Gelman

## Details

The following line of text is announcing "VA" as "Virginia" and not "Vee-Ay" like other texts.

`Other ways to get answers from VA`

When using abbreviations, screen readers can't always figure out how it should be pronounced.

![virginia](https://user-images.githubusercontent.com/984252/108730791-b8642800-74e0-11eb-8d5b-f782ce128789.PNG)

## Acceptance Criteria

- [x] Add aria-label to h2 correctly announce it as "Veterans Affairs" or "VA"

## Proposed Solution (if known)

`<h2 id="other-ways-to-get-answersfrom-" aria-label="Other ways to get answers from V A">Other ways to get answers from VA</h2>`

**Comment by RLHecht** [comment]
@noahgelman Can you provide more information for this ticket? I'm not sure what page I'm supposed to make changes on. Thanks!

**Comment by noahgelman** [comment]
@RLHecht Elaborated a bit more in detail in the "Details" section

**Comment by RLHecht** [comment]
@brianalloyd This is would be a sitewide FE issue. Passing over to your team.

**Comment by noahgelman** [comment]
@RLHecht I'm not sure if this is global. VA usually announces correctly. In this particular case the word "from" before "VA" makes the screen reader think we're talking about a physical location, and it guesses "Virginia".

Maybe it is a sitewide issue. You all can decide that.

**Comment by RLHecht** [comment]
Got it. We could add the aria label to just this text for now. Though could someone tell me exactly what that would look like and I'll add it in Drupal?

**Comment by brianalloyd** [comment]
@kelsonic looking into sitewide impact.  @RLHecht we'll follow up on updates that might be required sitewide.

**Comment by ncksllvn** [comment]
I am gathering that this is site-wide in the sense that a screenreader will encounter this issue anywhere it sees "_from_ VA." I am seeing a bunch of instances of this happening when I search the CMS content (almost 200 cases.) I don't think this task can be automated in any easy way because it is fully possibly that in some (unlikely but plausible) cases we _do_ mean Virginia instead of the VA. This means that a CMS editor will need to manually apply the fix described in this ticket to every instance. 

@noahgelman ^^ does that chime with your thoughts as well? Or does this issue possibly only occur inside of heading tags?

So here is what I am wondering -

1. Do we want this ticket to be regarded as a site-wide issue? If so, we need to plan a discussion with CSM on what (if anything) can be done to help automate this, or if we will need to just document it (eg. when saying "from VA" make sure to use an aria-label around VA to avoid mixup with Virginia) in the design system or something. Also, if we intend for this ticket to address every instance across the site, let's pop that estimate up to something mighty like an 8 because an engineer or two will need to offer support to a content editor. And we should groom the issue description again to make sure the acceptance criteria explicitly calls for capturing every instance.
1. Or we can scope this issue to just address what content shown in the screenshots of the issue description, which is a pic of the heading on `/contact-us`. If we want to do that, we can probably have an editor fix this really easily in the CMS.

Personally I think this issue should call for option 2. `/contact-us` is a relatively recent addition to the site so I think it was intending to address just that page. 

@noahgelman @brianalloyd thoughts?

**Comment by ncksllvn** [comment]
Went ahead and reached out to the CMS team for support, https://dsva.slack.com/archives/CDHBKAL9W/p1614869476058300

**Comment by swirtSJW** [comment]
The other issue is that VA is also a brand
![image](https://user-images.githubusercontent.com/5752113/109987850-fa594f00-7cd4-11eb-93f4-eb6ff4fd6b25.png)

https://newscdn2.weigelbroadcasting.com/PNA8j-1538574863-tile_image-120148.jpg

This solution would be inserting words where none exist.

**Comment by ncksllvn** [comment]
> Honestly I think this is a case of the screenreader overreaching with the assumption. Is this all screen readers making this assumption or just one?

Agree on SR overreaching. Using VoiceOver just now on `https://www.va.gov/contact-us/` it read `Other ways to get answers from VA` correctly as `Other ways to get answers from V-A`. @noahgelman is this issue w/JAWS specifically?  Or what SR?

**Comment by swirtSJW** [comment]
I just tried it with Google Screen reader and it read it fine.`Other ways to get answers from Vee Ayyy`.

**Comment by noahgelman** [comment]
@ncksllvn NVDA Windows Chrome

**Comment by artsymartha68** [comment]
This may be a dup of https://github.com/department-of-veterans-affairs/va.gov-team/issues/4507

**Comment by swirtSJW** [comment]
I think it is related to 4507 but not a duplicate.   Two different end user experiences, but could be resolved the same way.

**Comment by ncksllvn** [comment]
> This may be a dup of #4507

Definitely related....

I feel like we are inching towards "VA" always being wrapped with an `<abbr>` tag on the website. What do you all think?

**Comment by artsymartha68** [comment]
Can we get @1Copenut to take a look at a prototype with the <abbr> tag and see if it works better?

**Comment by ncksllvn** [comment]
> Can we get @1Copenut to take a look at a prototype with the tag and see if it works better?

Is it possible to have @1Copenut perform that change himself using his browser's web inspector? Modifying HTML locally is a pretty trivial task for a FE engineer and would save us on the back and forth. 

Example on `/contact-us/` - I inspected the HTML and edited locally using web tools

![image](https://user-images.githubusercontent.com/1915775/110132408-22aa8180-7d99-11eb-9a2f-92f7c178ff94.png)



cc @brianalloyd after this discussion unfolding I am bumping estimate to an 8 because an actionable path forward is unclear.

**Comment by brianalloyd** [comment]
@ncksllvn will follow up on this ticket and determine path forward.

**Comment by kelsonic** [comment]
@brianalloyd Should we move this into blocked until this ticket has a clear implementation path?

**Comment by brianalloyd** [comment]
@kelsonic great reminder, intended to do this yesterday post planning meeting.  Shifted to blocked as we explore clear path options.  Please to revisit in grooming for solution ideation.

**Comment by ncksllvn** [comment]
I propose we look at this issue in isolation instead of trying to do something site-wide. Site-wide considerations can be considered later in https://github.com/department-of-veterans-affairs/va.gov-team/issues/4507.

@swirtSJW is the `abbr` tag supported in this heading?

1. Visit https://prod.cms.va.gov/contact-us
2. Change heading `Other ways to get answers from VA` to `Other ways to get answers from <abbr title="Veterans Affairs">VA</abbr>`

**Comment by swirtSJW** [comment]
@ncksllvn At present, no that tag is not supported.   We could and should add it as allowed tag and attribute within the WYSIWYG in the CMS.  It would not be a heavy lift to do it.

**Comment by swirtSJW** [comment]
I just created this on the CMS side https://github.com/department-of-veterans-affairs/va.gov-cms/issues/4920  adding the abbr and acronym tag (this case should use acronym)  https://github.com/department-of-veterans-affairs/va.gov-cms/issues/4920

**Comment by swirtSJW** [comment]
Once we add it, then the content could be updated.   However I do think there is some governance in put needed related to VA being a brand that may need phonetic 'Vee Ay" rathern than "Veterans Administration.

**Comment by jenniferlee-dsva** [comment]
Thanks for thinking of the governance aspect @swirtSJW - really appreciate that! 

@brianalloyd @RLHecht cc @johnhashva - for screen reader aria labels on the acronym "VA," I think we should use the pronunciation of the term as an acronym - i.e.,  V ( as in the letter) + A (as in the letter)   or 'vee-ay' as steve put it, rather than replacing it with "Veterans Administration." 

I think because the branded VA term can sometimes be used as a noun and sometimes as an adjective within a sentence -- and the different ways that we call for [the sentence construction based on noun vs adj](https://design.va.gov/content-style-guide/word-list#v) - we should  stick closely to the intended acronym pronunciation to prevent any unexpected weirdness with screen readers. 

Also, we sometimes use the VA branded acronym as part of "VA.gov" -- and we wouldn't want to accidentally have screen readers 'translating' that as 'veterans-affairs-dot-gov.'   I don't know that it would, but just in case that's a possibility.

**Comment by artsymartha68** [comment]
I would love to get @AngelaFowler82 's input as a screenreader user. 

From a branding perspective, "Vee A" seems to be the way we should pronounce VA (plus it's so much shorter than "Veterans Administration").

**Comment by AngelaFowler82** [comment]
Screen readers do this on occasion, and we usually write it off as equal parts amusing and annoying, but naming a specific state on a VA web site may cause real confusion. I'd like to do some testing with JAWS, NVDA and Voiceover on the iPhone. On which URL is this occurring?

**Comment by kelsonic** [comment]
Hey @joshkimux @1Copenut! 👋 

I've been looking at this ticket as well as [this related ticket about the word FAQ](https://github.com/department-of-veterans-affairs/va.gov-team/issues/20237) and I have an idea on how to solve this, but it comes with some consequences.

Essentially, I can create a post-build script that goes through every generated HTML file and find/replace `FAQ` + `VA` with a wrapping span tag that has an `aria-label` attribute. I would create the file as a sibling to [this file](https://github.com/department-of-veterans-affairs/content-build/blob/bb699b60ead87720374c6f56630f6555051819d6/src/site/stages/build/plugins/modify-dom/update-external-links.js).

This would change replace all instances of `FAQ` and `VA` across va.gov with the wrapping span tag + aria-label attribute, but doing this will also (perhaps significantly) increase build times in CI since it will need to be executed at the end of each build. Do you have thoughts/suggestions regarding this idea?

I'm also guessing that screen reader users must be used to hearing "FAQ" pronounced as is on other websites since that abbreviation is used a lot across the internet -- not sure about "VA" but it could also be familiar for users who visit our website frequently.

**Comment by joshkimux** [comment]
@kelsonic , don't worry about it in this instance. Folks will be familiar with the abbreviation FAQ, and there almost always will be a fully spelled out "Frequently Asked Questions" nearby within the content.

**Comment by kelsonic** [comment]
Sounds great, thanks @joshkimux! What about "VA"?

**Comment by joshkimux** [comment]
@kelsonic , that one's more tricky as I'm not sure what unintended consequences may emerge-- I'd recommend we get more eyes on it from multiple perspectives to make sure it's the right solution.

**Comment by noahgelman** [comment]
@kelsonic 
As long as it's run once to clear out any previous instances, we can make sure to check for any new instances in our QA process to make sure it's done manually

**Comment by kelsonic** [comment]
Hey @noahgelman! Can content editors add custom markup in their HTML (e.g. `<span aria-label="">`) without it being sanitized out? I had thought we were sanitizing out HTML attributes in Drupal, I believe @RLHecht mentioned that a few times 😭

**Comment by brianalloyd** [comment]
@joshkimux @noahgelman we did some work on this but to completely fix the issue it would be a massive lift since it would require us to go through all pages and replace VA with `<span aria-label="Veterans Affairs">VA</span>` for all instances of VA across the site.  I wanted to validate intended outcome and might need to move this in backlog until our team has bandwidth in 2022.

**Comment by laflannery** [comment]
Closing as this is an older issue and when retesting is not happening on NVDA anymore
